code/game/mecha/mecha_actions.dm

```diff
@@ -19,13 +19,17 @@
 	internals_action.Grant(user, src)
 	lights_action.Grant(user, src)
 	stats_action.Grant(user, src)
+	for(var/obj/item/mecha_parts/mecha_equipment/thrusters/thrusters in equipment)
+		add_thrusters()
 
 /obj/mecha/proc/RemoveActions(mob/living/user, human_occupant = 0)
 	if(human_occupant)
 		eject_action.Remove(user)
 	internals_action.Remove(user)
 	lights_action.Remove(user)
 	stats_action.Remove(user)
+	for(var/obj/item/mecha_parts/mecha_equipment/thrusters/thrusters in equipment)
```

Comment:
should this be an `if` instead of a `for` ?

Reply:
A locate call has been put here